src/Avalonia.Visuals/Media/Immutable/ImmutableSolidColorBrush.cs

```diff
@@ -55,5 +55,32 @@ public override string ToString()
         {
             return Color.ToString();
         }
+
+        /// <inheritdoc/>
+        public bool Equals(ISolidColorBrush other)
+        {
+            return other != null
+                ? Hash(this) == Hash(other)
+                : false;
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return obj is ISolidColorBrush other
+                ? Equals(other)
+                : false;
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return Hash(this);
+        }
+
+        private static int Hash(ISolidColorBrush brush)
+        {
+            return (brush.Color, brush.Opacity).GetHashCode();
```

Comment:
Might it be better to make `SolidColorBrush.Hash` `internal` and call that? It's probably not a big deal for `SolidColorBrush` but for other brushes with more complex logic we probably don't want to be duplicating the code.

Reply:
I dropped the whole method because its now only used in GetHashCode